scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,107 @@
+"""
+This script is referenced by TerraForm as an external data source. It prints the
+current deployment's Elasticsearch node IDs as JSON to STDOUT. If the list of
+node IDs is less than the number of placeholders in the Cloudwatch dashboard
+template, the output is padded with 'blank' entries.
+Example output:
+{"nodes": "tVsGH2BSXXYe9sMLEYRcIv,ydG9meZMAGkTnAlPgLljz8,blank,blank"}
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+import re
+from time import (
+    sleep,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+import jq
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to retry the request to AWS if an error is encountered.
+max_attempts = 10
+
+
+def main() -> None:
+    for _ in range(max_attempts):
+        try:
+            response = aws.cloudwatch.list_metrics(Namespace='AWS/ES',
+                                                   MetricName='CPUUtilization',
+                                                   RecentlyActive='PT3H',
+                                                   Dimensions=[
+                                                       {
+                                                           'Name': 'DomainName',
+                                                           'Value': config.es_domain
+                                                       }
+                                                   ])
+        except ClientError as e:
+            log.debug('Error: ' + e.response['Error']['Code'] + '. Retrying...')
+        else:
+            nodes = [
+                d['Value']
+                for m in response['Metrics']
+                for d in m['Dimensions']
+                if d['Name'] == 'NodeId'
+            ]
+            if len(nodes) > 0:
+                break
```

Comment:
Extract function, 

```suggestion
                return nodes
```

and eliminate `else` on loop.